db/migrate/20241104161404_add_competing_status_enum.rb

```diff
@@ -0,0 +1,12 @@
+# frozen_string_literal: true
+
+class AddCompetingStatusEnum < ActiveRecord::Migration[7.2]
+  def up
+    add_column :registrations, :competing_status, :string, default: Registrations::Helper::STATUS_PENDING, null: false
+
+    # Update values based on the old boolean column
+    Registration.all.each do |r|
```

Comment:
For bulk updates like this, that cannot be done in pure SQL, you can use `Registration.find_each`. It's more memory-efficient by batching the model load